### DISCUSSION TITLE
[NUXT 3] API call on every route change

### DISCUSSION THREAD

**Discussion by mathishuettl** [nuxt/nuxt#17248] [Questions]
Hello,

I got a [...slug].vue file, that should catch every page, in the setup function I do an API-Call to get the specific page from my API, the problem is, that when i change the page (click on a ```<NuxtLink>```) nothing happens, so the data are still the same.

I already tried to manage this with watching the $route, but it didn't help...

What I want to archive is the following:
When navigating on the Website I want to do an API Request on each

``` javascript
<template>
    <div>
        <p>Slug: {{ $route.params.slug }}</p>
        <p v-if="page && page.post_title">Title: {{ page.post_title }}</p>
    </div>
</template>


<script setup>
    const route = useRoute()
    const { data:page } = await useAsyncData("page", () => $fetch(`http://nuxt.local/wp-json/custom-theme/v1/route?url=${route.path}`))    
</script>

<script>
    export default {
        methods: {
           
        },
        watch: {
            $route (to, from) {
                $fetch(`http://nuxt.local/wp-json/custom-theme/v1/route?url=${this.$route.path}`).then(response => {
                    this.page = response
                })
            },
        }
    }
</script>
``` 

I also tried arround with the refresh() method, returned by useFetch, but it acts really strange (see video) => on the first few clicks the title doesn't change, but suddenly it does.. 

```javascript
<script setup>
    const route = useRoute()
    const { data:page, refresh } = await useFetch(`http://nuxt.local/wp-json/custom-theme/v1/route?url=${route.path}`)
    console.log(page.page_title)
</script>

<script>
    export default {
        methods: {
           
        },
        watch: {
            $route (to, from) {
                this.refresh()
            },
        }
    }
</script>
```

Here is a video... it acts really strange and i cant see a pattern... https://www.veed.io/view/5febbf36-415e-4390-a7fb-30deeeba99eb

**Comment by ExEr7um** [comment]
Try using `initialCache: false`.
Something like this:
```js
const { data: response } = await useFetch<BlogPost>(
  `api/endpoint`,
  {
    initialCache: false
  }
)
```

**Reply by mathishuettl** [reply]
doesn't help.. i tried it with the following: 

```javascript
<script setup>
    const route = useRoute()
    const { data:page, refresh } = await useFetch(`http://nuxt.local/wp-json/custom-theme/v1/route?url=${route.path}`, {
        initialCache: false
    })
</script>
```

**Reply by mathishuettl** [reply]
I just created a small screencapture to show you how it acts: https://www.veed.io/view/5febbf36-415e-4390-a7fb-30deeeba99eb

**Reply by ExEr7um** [reply]
Maybe I found a solution. I use `useAsyncData` instead of `useFetch` and provide it with unique request id `nuxt-local-${route.path}`. Something like this:
```vue
<script setup>
const route = useRoute()
const { data: page } = await useAsyncData(`nuxt-local-${route.path}`, () => $fetch(`http://nuxt.local/wp-json/custom-theme/v1/route?url=${route.path}`))
</script>
```
